pandas/tests/test_window.py

```diff
@@ -3356,6 +3356,34 @@ def test_min_periods(self):
         result = df.rolling('2s', min_periods=1).sum()
         tm.assert_frame_equal(result, expected)
 
+    def test_closed(self):
+        df = DataFrame({'A': [1]*5},
+                        index = [pd.Timestamp('20130101 09:00:01'),
+                                 pd.Timestamp('20130101 09:00:02'),
+                                 pd.Timestamp('20130101 09:00:03'),
+                                 pd.Timestamp('20130101 09:00:04'),
+                                 pd.Timestamp('20130101 09:00:06')])
+
+        expected = df.copy()
+        expected["A"] = [1.0, 2, 2, 2, 1]
+        result = df.rolling('2s', closed='right').sum()
+        tm.assert_frame_equal(result, expected)
+
+        expected = df.copy()
+        expected["A"] = [1.0, 2, 3, 3, 2]
+        result = df.rolling('2s', closed='both').sum()
+        tm.assert_frame_equal(result, expected)
+
+        expected = df.copy()
+        expected["A"] = [np.nan, 1.0, 2, 2, 1]
+        result = df.rolling('2s', closed='left').sum()
+        tm.assert_frame_equal(result, expected)
+
```

Comment:
If the DataFrame is empty, the behavior is exactly the same as in the current implementation. For time-based windows, it throws "ValueError: window must be an integer".  Note that this feature only affects the window indexer's start and end vectors. In fixed windows, empty DataFrames always return empty (the result is constructed as np.empty(N) where N is the length of the input)